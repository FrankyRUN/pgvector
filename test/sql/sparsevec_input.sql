SELECT '{0:1.5,2:3.5}/5'::sparsevec;
SELECT '{0:1.5,2:3.5}/5'::sparsevec::vector;
SELECT '{0:1.5,2:3.5}/5'::sparsevec::vector(5);
SELECT '{0:1.5,2:3.5}/5'::sparsevec::vector(4);
SELECT '[0,1.5,0,3.5,0]'::vector::sparsevec;

SELECT '{0:0,1:1,2:0}/3'::sparsevec;

SELECT '{1:1,0:1}/2'::sparsevec;

SELECT '{}/5'::sparsevec;
SELECT '{}/-1'::sparsevec;
SELECT '{}/100001'::sparsevec;
SELECT '{}/16001'::sparsevec::vector;

SELECT '{-1:1}/1'::sparsevec;
SELECT '{1:1}/1'::sparsevec;

SELECT '{}/1'::sparsevec(2);
