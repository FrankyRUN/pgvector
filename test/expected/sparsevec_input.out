SELECT '{0:1.5,2:3.5}/5'::sparsevec;
    sparsevec    
-----------------
 {0:1.5,2:3.5}/5
(1 row)

SELECT '{0:1.5,2:3.5}/5'::sparsevec::vector;
     vector      
-----------------
 [1.5,0,3.5,0,0]
(1 row)

SELECT '{0:1.5,2:3.5}/5'::sparsevec::vector(5);
     vector      
-----------------
 [1.5,0,3.5,0,0]
(1 row)

SELECT '{0:1.5,2:3.5}/5'::sparsevec::vector(4);
ERROR:  expected 4 dimensions, not 5
SELECT '[0,1.5,0,3.5,0]'::vector::sparsevec;
    sparsevec    
-----------------
 {1:1.5,3:3.5}/5
(1 row)

SELECT '{0:0,1:1,2:0}/3'::sparsevec;
 sparsevec 
-----------
 {1:1}/3
(1 row)

SELECT '{1:1,0:1}/2'::sparsevec;
ERROR:  indexes must be in ascending order
LINE 1: SELECT '{1:1,0:1}/2'::sparsevec;
               ^
SELECT '{}/5'::sparsevec;
 sparsevec 
-----------
 {}/5
(1 row)

SELECT '{}/-1'::sparsevec;
ERROR:  sparsevec must have at least 1 dimension
LINE 1: SELECT '{}/-1'::sparsevec;
               ^
SELECT '{}/100001'::sparsevec;
ERROR:  sparsevec cannot have more than 100000 dimensions
LINE 1: SELECT '{}/100001'::sparsevec;
               ^
SELECT '{}/16001'::sparsevec::vector;
ERROR:  vector cannot have more than 16000 dimensions
SELECT '{-1:1}/1'::sparsevec;
ERROR:  index "-1" is out of range for type sparsevec
LINE 1: SELECT '{-1:1}/1'::sparsevec;
               ^
SELECT '{1:1}/1'::sparsevec;
ERROR:  index must be less than dimensions
LINE 1: SELECT '{1:1}/1'::sparsevec;
               ^
SELECT '{}/1'::sparsevec(2);
ERROR:  expected 2 dimensions, not 1
